src/structures/GuildMember.js

```diff
@@ -337,6 +346,22 @@ class GuildMember extends Base {
     return this.guild.members.ban(this, options);
   }
 
+  /**
+   * Timeouts this guild member.
+   * @param {TimeoutDateResolvable|null} timeout The date, timestamp, or time in seconds
+   * for the member's communication to be disabled until. Provide `null` to remove the timeout.
+   * @param {string} [reason] The reason for this timeout.
+   * @returns {Promise<GuildMember>}
+   * @example
+   * // Timeout a guild member for 5 minutes
+   * guildMember.timeout(300, 'They deserved it')
+   *   .then(console.log)
+   *   .catch(console.error);
+   */
+  timeout(timeout, reason) {
```

Comment:
This is the client name, don't we usually follow the API name?
```suggestion
  disableCommunication(timeout, reason) {
```

Reply:
Won't it be better to use a object instead of multiply parameters, would be more consistence to `.ban` ...

Reply:
The API calls are usually done by the manager by resolving the respective id, collecting all the JSON query data in an object followed by additional headers (`reason` typically). Sometimes, there won't be a `reason` and helper methods do not need an `id` - they are called from the structure itself.

This results in the cases of helper methods _usually_ having 2 parameters: the data to send across and the reason. If there's no reason, then it's not included. If there's no need for any data, then there won't be any parameters. The [`Guild`](https://discord.js.org/#/docs/main/main/class/Guild) class is a prime example of this.

`<GuildMember>.ban()` is the inconsistent one: providing a `reason` was part of the JSON query data which is likely why it was included in the type definition, but this is deprecated as of a while ago and has since been moved out of the JSON query data sent to Discord (but still in the type definition). It is simply still reused across the library.

I would also argue that anything that doesn't follow this pattern is inconsistent, such as [`GuildStickerManager#create()`](https://discord.js.org/#/docs/main/main/class/GuildStickerManager?scrollTo=create) which seems off to be honest... 4 possible parameters instead of 1 object with a `reason` parameter?

Reply:
> This is the client name, don't we usually follow the API name?

I'd like to think that helper methods help bridge the gap between the client and the API. That's why I went for "timeout" here. It's known to be the timeout feature so I'm sure that's what people expect to see! Welcome to hear feedback (especially from maintainers) about whether to change it back.

Reply:
There are other methods such as `setRateLimitPerUser` that have a different name in the API and the client and yet Discord.js went with the API name for all methods and properties so I think it would make sense to do the same here, as I don't recall this being done anywhere else other than in JSDocs and this would introduce a weird inconsistency in the library

Reply:
We should probably name this either `timeoutUntil`, `setTimeoutUntil`, or `disableCommunicationUntil` since it takes a date or an UNIX timestamp, and not a duration.

But I'm more inclined for `disableCommunicationUntil` because timeouts remind us of the `setTimeout` global, which takes a callback and a duration, which neither of which this method takes.

Reply:
Just to be clear, you're moving towards "disableCommunicationUntil" with no duration?

Reply:
If we take a duration, `timeout` and `disableCommunicationFor` are fine.

Also, why not both? cc @discordjs/the-big-4

Reply:
Just to be clear, this helper method already accepts a date and a time in milliseconds (duration).

Reply:
Honestly, I'm also much more of a fan of the `disableCommunicationUntil` name for a method that acts just as a shortcut, and something like `timeout` or w/e that gives you the support for passing in a duration

Reply:
See bab6adc64bbe57ba6cbb74ff3ba69d5de96a82c4!